server/src/main/java/org/elasticsearch/index/query/SearchExecutionContext.java

```diff
@@ -106,7 +111,7 @@ public SearchExecutionContext(
         int shardRequestIndex,
         IndexSettings indexSettings,
         BitsetFilterCache bitsetFilterCache,
-        BiFunction<MappedFieldType, FieldDataContext, IndexFieldData<?>> indexFieldDataLookup,
+        IndexFieldDataLookup indexFieldDataLookup,
```

Comment:
I thought that introducing an interface here instead of adding another `BiFunction` would have made things clearer, but I have second thoughts about it. See `SearchLookup` for contrast, where instead I introduced an additional function